src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,400 @@
+---
+
+> *Since v0.29.0 (JavaScript extensions) and v0.31.0 (Output extensions)*
+
+Traditionally, extending k6 with custom functionality that isn't available in the
+open source tool has been possible in one of two ways:
+
+- By importing a JavaScript library. This is a simple way to extend the functionality
+  of a test script, though it has some drawbacks. Since JavaScript in k6 runs in a
+  virtual machine [that is unlike the one used in web browsers and
+  NodeJS](/#what-k6-does-not), it's not possible to use libraries that write to
+  disk, use JS APIs other than the ones built into k6, or need some new binary
+  protocol support. Furthermore, while k6 performs well in most use cases, all
+  execution is interpreted by the [Go JavaScript runtime](https://github.com/dop251/goja),
+  which can impact the performance of certain resource intensive operations
+  (cryptography, compression, etc.).
+
+- By forking the [k6 repository](https://github.com/k6io/k6), doing the changes in Go
+  and submitting a pull request for review by the core team (preferably following
+  the [contribution guidelines](https://github.com/k6io/k6/blob/master/CONTRIBUTING.md)).
+  This is a great way to contribute new core features, but it can be a lengthy
+  process, and submissions might be rejected if they don't align with the long-term
+  vision of the project.
+
+To address these issues and allow the community to more easily adapt k6 to fit their
+needs, [k6 v0.29.0](https://github.com/k6io/k6/releases/tag/v0.29.0) introduced
+the [xk6 framework](https://github.com/k6io/xk6) and the concept of k6 extensions.
+
+
+## What are k6 extensions?
+
+k6 extensions are standalone Go projects that call k6 APIs but are otherwise
+unrestricted in their functionality. This provides the freedom for extension authors
+to experiment with novel integrations with k6 that could eventually easily become
+part of core. In this sense extensions can be thought of as a "testing ground" for
+eventual promotion upstream, once their features and API is stable, and given that
+the extension is generally useful for all k6 users.
+
+
+## What is xk6?
+
+[xk6](https://github.com/k6io/xk6) is a command-line tool and framework inspired by
+[xcaddy](https://github.com/caddyserver/xcaddy), designed for building custom k6
+binaries that bundle one or more extensions written in Go.
+
+Its main features are:
+
+- Ease of use: with a few commands even less technical users should be able to build
+  their own k6 binaries, given that they have the Go toolchain installed and any
+  dependencies required by a specific extension.
+- Simple API for Go programmers that handles the Go<->JS translation, with the
+  ability to call any public k6 Go API. Extensions are first-class components along
+  with other built-in modules.
+- Cross-platform like Go and runs great on macOS, Windows and Linux.
+
+
+### Extension types
+
+The initial version of xk6 released in v0.29.0 supported only JavaScript extensions,
```

Comment:
imo, it is unnecessary to mention again the 0.29 / 0.31 versions

Reply:
OK, done in 0ec0f65.